plugins/scope/request_scope.go

```diff
@@ -0,0 +1,33 @@
+package scope
+
+import (
+	"net/http"
+
+	"github.com/google/uuid"
+	"go.mondoo.com/ranger-rpc"
+)
+
+const XRequestID = "X-Request-ID"
+
+type requestIdPlugin struct {
+	requestID string
+}
+
+// NewRequestIDPlugin creates a RangerPlugin that adds the
+// X-Request-ID header to outgoing requests. All RPCs
+// that share the same RangerPlugin will get the same ID
+func NewRequestIDRangerPlugin() ranger.ClientPlugin {
```

Comment:
Maybe we could also add another function that allows injecting the uuid? Could be useful for testing purposes
```
func NewRequestIDRangerPlugin() ranger.ClientPlugin {
id := uuid.new().string()
	return NewRequestIDRangerPlugin(id)
}

func NewRequestIDRangerPlugin(reqId string) ranger.ClientPlugin {
	return &requestIdPlugin{
		requestID: reqId
	}
}

Reply:
Good idea, I'll add an optional flag to `NewRequestIDRangerPlugin`

Reply:
solved with new optional argument, test added as well